libp2p-swarm/src/connection_reuse.rs

```diff
@@ -127,31 +142,36 @@ where
 	}
 
 	fn dial(self, addr: Multiaddr) -> Result<Self::Dial, (Self, Multiaddr)> {
```

Comment:
what does `Err` variant represent?  and in what cases is it returned?

Reply:
Happens if the multiaddress is invalid or not supported. This should be documented in the `Transport` trait if it isn't the case.